lib/stdio/lines.js

```diff
@@ -0,0 +1,57 @@
+// Split chunks line-wise
+export const getLinesGenerator = (encoding, binary) => {
+	if (binary) {
+		return;
+	}
+
+	return encoding === 'buffer' ? linesUint8ArrayGenerator : linesStringGenerator;
+};
+
+const linesUint8ArrayGenerator = async function * (chunks) {
+	yield * linesGenerator(chunks, new Uint8Array(0), 0x0A, concatUint8Array);
+};
+
+const concatUint8Array = (firstChunk, secondChunk) => {
+	const chunk = new Uint8Array(firstChunk.length + secondChunk.length);
+	chunk.set(firstChunk, 0);
+	chunk.set(secondChunk, firstChunk.length);
+	return chunk;
+};
+
+const linesStringGenerator = async function * (chunks) {
+	yield * linesGenerator(chunks, '', '\n', concatString);
+};
+
+const concatString = (firstChunk, secondChunk) => `${firstChunk}${secondChunk}`;
+
+// This imperative logic is much faster than using `String.split()` and uses very low memory.
```

Comment:
Quick benchmark on my machine was showing this code to be ~6 times faster than `String.split()`.